deconstrst/builders/single.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+
+import os
+import mimetypes
+import json
+from os import path
+
+import requests
+from docutils import nodes
+from docutils.io import DocTreeInput, StringOutput
+from sphinx.builders.html import SingleFileHTMLBuilder
+from sphinx.builders.html import JSONHTMLBuilder
+from sphinx.util import jsonimpl
+from sphinx.util.osutil import os_path, relative_uri
+from sphinx.util.console import bold, darkgreen, brown
+from sphinx.writers.html import HTMLWriter
+from sphinx.config import Config
+from deconstrst.config import Configuration
+from pprint import pprint
+
+# Tell Sphinx about the deconst_default_layout key.
+Config.config_values["deconst_default_layout"] = ("default", "html")
+
+
+class DeconstSingleJSONBuilder(SingleFileHTMLBuilder):
+    """
+    Custom Sphinx builder that generates Deconst-compatible JSON documents.
+    """
+
+    name = 'deconst-single'
+
+    def init(self):
+        SingleFileHTMLBuilder.init(self)
+
+        self.deconst_config = Configuration(os.environ)
+
+        if os.path.exists("_deconst.json"):
+            with open("_deconst.json", "r") as cf:
+                self.deconst_config.apply_file(cf)
```

Comment:
I really should wrap this up in the `Configuration` constructor, so you don't have to remember to call it manually every time you get a `Configuration` object.

Also, will this read from the wrong place if you specify a different source directory?